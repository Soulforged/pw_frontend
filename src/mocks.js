//@flow

//FIXME remove all mocks from here when they're no longer necessary
/* eslint-disable global-require,import/prefer-default-export */
export default {
  summary: require("./data/summary.json"),
  trendsByDate: require("./data/trendsByDate.json"),
  trendsByInstitution: require("./data/trendsByInstitution.json"),
  errors: require("./data/errors.json"),
  users: require("./data/users.json"),
  login: require("./data/login.json"),
  user: require("./data/user.json"),
  userCreate: require("./data/userCreate.json"),
  role: require("./data/role.json")
};

/* eslint-disable */
