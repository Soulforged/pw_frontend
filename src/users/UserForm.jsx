//@flow
import React from "react";
import { Text, Select } from "react-form";
import { CRUDForm } from "src/components";

const statusOptions = [
  { label: "Active", value: 1 },
  { label: "Inactive", value: 0 },
];

const roleOptions = [
  { label: "Choose a role", value: null },
  { label: "A role", value: 1 }
];

type Props = {
  item: Object,
  saveUser: (Object) => void,
  saving: boolean,
  returnToList: () => void,
  fetchUser: (number) => void
};

export default (props: Props) => (
  <CRUDForm
    save={props.saveUser}
    entityName="user"
    onClose={props.returnToList}
    loader={props.fetchUser}
    {...props}
  >
    <div className="row">
      <label htmlFor="firstName">
        First Name:
        <Text
          field="firstName"
          id="firstName"
          className="form-control"
          placeholder="First name"
          required
        />
      </label>
    </div>
    <div className="row">
      <label htmlFor="lastName">
        Last Name:
        <Text
          id="lastName"
          className="form-control"
          field="lastName"
          placeholder="Last name"
          required
        />
      </label>
    </div>
    <div className="row">
      <label htmlFor="userName">
        Username:
        <Text
          id="userName"
          className="form-control"
          field="userName"
          placeholder="Username"
          required
        />
      </label>
    </div>
    <div className="row">
      <label htmlFor="mobileNumber">
        Phone number:
        <Text
          id="mobileNumber"
          className="form-control"
          field="mobileNumber"
          placeholder="Mobile phone number"
          required
        />
      </label>
    </div>
    <div className="row">
      <label htmlFor="primaryEmail">
        Email:
        <Text
          id="primaryEmail"
          className="form-control"
          field="primaryEmail"
          placeholder="Email (Optional)"
        />
      </label>
    </div>
    <div className="row">
      <label htmlFor="status">
        Status
        <Select
          id="status"
          className="form-control"
          field="status"
          options={statusOptions}
        />
      </label>
    </div>
    <div className="row">
      <label htmlFor="businessUnitName">
        Business Unit
        <Select
          id="businessUnitName"
          className="form-control"
          field="businessUnitName"
          options={[{ label: "Choose a business unit", value: null }]}
        />
      </label>
    </div>
    <div className="row">
      <label htmlFor="roleId">
        Role
        <Select
          id="roleId"
          className="form-control"
          field="roleId"
          options={roleOptions}
        />
      </label>
    </div>
  </CRUDForm>
);
