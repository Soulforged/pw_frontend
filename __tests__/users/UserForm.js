//@flow
import React from "react";
import UserForm from "src/users/UserForm";
import renderer from "react-test-renderer";
import mocks from "src/mocks";

it("matches snapshot", () => {
  const rendered = renderer.create(<UserForm fetchUser={jest.fn()} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
