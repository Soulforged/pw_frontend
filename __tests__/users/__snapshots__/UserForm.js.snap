// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot no data 1`] = `
<div
  className="pnls"
  id="main-pnl"
>
  Not found
</div>
`;

exports[`matches snapshot while fetching 1`] = `
<div
  id="loader"
  style={Object {}}
>
  <img
    alt="loading something..."
    className="loader"
    src="loader.gif"
  />
</div>
`;

exports[`matches snapshot while not saving 1`] = `
<div
  className="pnls"
  id="main-pnl"
>
  <h4
    className="form-header trebuchet text-center bold"
  >
    Edit user 117
    <button
      className="add-new pointer pull-right bold"
      onClick={undefined}
    >
      <i
        className="fa fa-close theme"
      />
    </button>
  </h4>
  <div
    className="add-pnl-cnt"
  >
    <form
      id="user_frm"
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <label
          htmlFor="firstName"
        >
          First Name:
          <input
            className="form-control"
            id="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="First name"
            required={true}
            value="SAMPLE"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="lastName"
        >
          Last Name:
          <input
            className="form-control"
            id="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Last name"
            required={true}
            value="DOE"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="userName"
        >
          Username:
          <input
            className="form-control"
            id="userName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Username"
            required={true}
            value="robert"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="mobileNumber"
        >
          Phone number:
          <input
            className="form-control"
            id="mobileNumber"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Mobile phone number"
            required={true}
            value="254700156952"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="primaryEmail"
        >
          Email:
          <input
            className="form-control"
            id="primaryEmail"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email (Optional)"
            value="robert@merchant8.com"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="status"
        >
          Status
          <select
            className="form-control"
            id="status"
            onBlur={[Function]}
            onChange={[Function]}
            value={1}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              Active
            </option>
            <option
              disabled={undefined}
              value={2}
            >
              Inactive
            </option>
          </select>
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="businessUnitName"
        >
          Business Unit
          <select
            className="form-control"
            id="businessUnitName"
            onBlur={[Function]}
            onChange={[Function]}
            value={0}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              Choose a business unit
            </option>
          </select>
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="roleId"
        >
          Role
          <select
            className="form-control"
            id="roleId"
            onBlur={[Function]}
            onChange={[Function]}
            value={0}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              Choose a role
            </option>
            <option
              disabled={undefined}
              value={2}
            >
              A role
            </option>
          </select>
        </label>
      </div>
      <hr />
      <div />
      <article
        className="row modal-p pull-right"
      >
        <button
          className="btn btn-default"
          type="submit"
        >
          SAVE
        </button>
      </article>
    </form>
  </div>
</div>
`;

exports[`matches snapshot while saving 1`] = `
<div
  className="pnls"
  id="main-pnl"
>
  <h4
    className="form-header trebuchet text-center bold"
  >
    Edit user 117
    <button
      className="add-new pointer pull-right bold"
      onClick={undefined}
    >
      <i
        className="fa fa-close theme"
      />
    </button>
  </h4>
  <div
    className="add-pnl-cnt"
  >
    <form
      id="user_frm"
      onSubmit={[Function]}
    >
      <div
        className="row"
      >
        <label
          htmlFor="firstName"
        >
          First Name:
          <input
            className="form-control"
            id="firstName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="First name"
            required={true}
            value="SAMPLE"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="lastName"
        >
          Last Name:
          <input
            className="form-control"
            id="lastName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Last name"
            required={true}
            value="DOE"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="userName"
        >
          Username:
          <input
            className="form-control"
            id="userName"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Username"
            required={true}
            value="robert"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="mobileNumber"
        >
          Phone number:
          <input
            className="form-control"
            id="mobileNumber"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Mobile phone number"
            required={true}
            value="254700156952"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="primaryEmail"
        >
          Email:
          <input
            className="form-control"
            id="primaryEmail"
            onBlur={[Function]}
            onChange={[Function]}
            placeholder="Email (Optional)"
            value="robert@merchant8.com"
          />
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="status"
        >
          Status
          <select
            className="form-control"
            id="status"
            onBlur={[Function]}
            onChange={[Function]}
            value={1}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              Active
            </option>
            <option
              disabled={undefined}
              value={2}
            >
              Inactive
            </option>
          </select>
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="businessUnitName"
        >
          Business Unit
          <select
            className="form-control"
            id="businessUnitName"
            onBlur={[Function]}
            onChange={[Function]}
            value={0}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              Choose a business unit
            </option>
          </select>
        </label>
      </div>
      <div
        className="row"
      >
        <label
          htmlFor="roleId"
        >
          Role
          <select
            className="form-control"
            id="roleId"
            onBlur={[Function]}
            onChange={[Function]}
            value={0}
          >
            <option
              disabled={true}
              value={0}
            >
              Select One...
            </option>
            <option
              disabled={undefined}
              value={1}
            >
              Choose a role
            </option>
            <option
              disabled={undefined}
              value={2}
            >
              A role
            </option>
          </select>
        </label>
      </div>
      <hr />
      <div />
      <article
        className="row modal-p pull-right"
      >
        <div
          id="loader"
          style={Object {}}
        >
          <img
            alt="loading something..."
            className="loader"
            src="loader.gif"
          />
        </div>
      </article>
    </form>
  </div>
</div>
`;
